context/db.go

```diff
@@ -167,9 +167,8 @@ func (smContext *SMContext) UnmarshalJSON(data []byte) error {
 
 	var dataPathInDBIf interface{}
 	var FirstDPNodeIf interface{}
-	var nilVal *UPTunnelInDB = nil
 	smContext.Tunnel = &UPTunnel{}
-	if &aux.Tunnel != nilVal {
+	if aux != nil {
```

Comment:
I think this change is incorrect (`aux` will never be `nil` because it is initialized in line 145). Given that `UPTunnelInDB` is a struct, I think what you need to compare here is that `Tunnel` is not empty, correct?

Reply:
you are right, aux will never be nil

Reply:
done